### ISSUE TITLE
Script / Automation not working for Steam Deck OLED

### ISSUE THREAD

**Issue by DaLuke95** [Nonary/ResolutionAutomation#84]
Hi, I hope @Nonary can assist here quickly.
I'm a complete amateur when it comes to scripting whatnot so its likely that the mistake is just a wrong setting.

I installed your script (fairly easy, ofc).
It seems to try to do smt but the resolution remains the same when streaming via Sunshine.

notes:
- Windows 10 (HDR enabled)
- Steam Deck OLED
- resolution wanted (as your rec.) 2560 x 1600 @ 90hz
- Sunshine updated to newest version
- Moonlight client is set to custom resolution 2560x1600 @ 90hz
- custom resolution in Nvidia Control Panel 3840 x 2400 @ 90hz progressive is set/added (although nvidia didnt like that and my screen looked crazy)
- i have not made any changes to the JSON file because I couldnt figure out what "uncomment" is or rather how to _use_ it

If its just related to that JSON file, could you kindly send me the code (for the whole file according to the preferred res. and hz) or something like that?

Thank you very much, this seems like a very useful tool and I hope you can help!

**Comment by Nonary** [comment]
Have you tried the AI Chatbot for help? It's free to use even for non ChatGPT + subscribers

https://chatgpt.com/g/g-2mF6ae5iI-resolution-matcher-support

**Comment by DaLuke95** [comment]
I've tried and as i thought, my json file was not configured correctly.
However, Nvidia showed me the resolution and refresh rate and my GPU (RTX 4070 Ti Super does support all that)
I've also adjusted the GPU scaling in the nvidia control panel.

This is the summary of GPT:

Issue Summary
The script processes the resolution override (2560x1600x90=3840x2160x120) but fails to apply the client resolution (2560x1600x90).
Instead, it defaults to the host resolution (3840x2160x120).
The desired resolution (2560x1600x90) is not found in the host's display modes during the override attempt.


I've restarted a few times and left my TV (LG C9) off and now it seems to be working, for now. 
I will further test and report here if there are more issues.
I was also thinking to buy a dummy plug. But my GPU just features one HDMI port so I'd be stuck on Displayport and I couldnt find info if Displayport Dummy Plugs support 2560x1600x90 with HDR enabled.

**Comment by Nonary** [comment]
Can you upload the log so I can look at it.

Also enable debug mode in the settings.json file

**Comment by DaLuke95** [comment]
[log_1734216676.txt](https://github.com/user-attachments/files/18137896/log_1734216676.txt)

Hey man, ofc! Debug enabled. Tried again - no dice. Here is the log file. If we can get  this working, I'll happily donate!

**Comment by Nonary** [comment]
Yeah, your host computer doesn't have 2560x1600x90 available, you need to add it using CRU or NVIDIA Control Panel

Otherwise, the next closest one would be 2560x1600x100

Which you can add as an override

2560x1600x90=2560x1600x100


Or preferably, use 3840 x 2400 x 90 which was not found in your host either in the script log.

If that doesn't work, you need to use a virtual display.

https://github.com/Nonary/documentation/wiki/DummyPlugs

**Comment by DaLuke95** [comment]
But it _is_ set in the nvdia control panel. The picture looks a bit fucked off when I'm choosing it but looks how it should on the Steam Deck
![2024-12-15 11_16_08-NVIDIA Control Panel](https://github.com/user-attachments/assets/23ada989-737e-455a-ba80-3be7458180b1)
![2](https://github.com/user-attachments/assets/6fca33f9-f509-4c06-8628-37f456a82cca)
![3](https://github.com/user-attachments/assets/8132a9f2-b6d7-4489-b99d-dffbe58ddef6)


But I've tried it again, with 90hz. Still no dice,
What I just dont understand is, why the script cannot find it. It is in the Nvdia Control Panel - sometimes the Windows display settings wont let me change res but I thought that was, because Nvdia overrides it in a way. Because it works a 100% when I'm changeing the res through Nvdia.

**Comment by DaLuke95** [comment]
I'm able to set it, without turning on the TV and just connecting via Moonlight and then it put it out. 
![PXL_20241215_104754405 MP](https://github.com/user-attachments/assets/0d7d947a-5c8c-417c-8663-ebe7ded505f9)

**Comment by DaLuke95** [comment]
I even tried setting the value to 2560x1600x120fps (and then lock it to the 90hz that are only available at the Steam Deck (but Moonlight client set to 120)
Still no dice. But that is a combo that doesnt need a "custom creation" in the control panel since its "natively" supported.
Whats funny about this process is, I am able to set this manually + when I then stop the connection it reverts back to my normal TV res 3840 x 2160 x 120. So the 2 Step in the script (reverting back) is working.
Its just not switching automatically through your script. 
What I'm thinking is, maybe the script is not looking into the correct category.
It should look into "PC" but what I think, it does, its only looking into "Ultra HD, HD, SD" for some reason. Because when I just started Moonlight on the Deck and not the TV the Moonlight client / the script referred to the "Ultra HD, HD, SD" section and 4k x 2k x 60hz which is the max on that setting. But not the one i'm normally using
![what the script does](https://github.com/user-attachments/assets/135caa31-6bad-45ff-9853-7af46f32b34c)
![normal](https://github.com/user-attachments/assets/19b3d1f2-fc60-4ad6-ad1e-c316b25b48fe)
[log_1734283530.txt](https://github.com/user-attachments/files/18141199/log_1734283530.txt)

**Comment by Nonary** [comment]
You’ll need to add it using CRU or set up a virtual display instead at this point since the Windows API is not recognizing those resolutions.

In order for the resolution to be recognized it has to be the same in both active and desktop resolution (visible under advanced display info in settings)

often Windows 11 will allow you to do something like 2560x1600x90 but downsample from a higher resolution (this would show a higher active signal and lower desktop signal)

**Comment by DaLuke95** [comment]
Aight, I bought a dummy plug now. I saw you've made another script called "Monitor swap automation" -> I'll try that and hopefully this dummy plug works as intended

**Comment by DaLuke95** [comment]
Dummypplug came. All the EDID editing and stuff worked but the scripts didnt work unfortunately.
I thank you very much for your help but eihter I'm cursed or just plain incompetent. 

Literally the only thing I wanted was, when I played via my Steam Deck that the res changes and reverts after disconnecting. Probably its not the script but nvidia and Big M's incompetence

**Comment by Nonary** [comment]
> Dummypplug came. All the EDID editing and stuff worked but the scripts didnt work unfortunately. I thank you very much for your help but eihter I'm cursed or just plain incompetent.
> 
> Literally the only thing I wanted was, when I played via my Steam Deck that the res changes and reverts after disconnecting. Probably its not the script but nvidia and Big M's incompetence

Then I guess you'll need to just use RTSS to frame cap and then have an override that will set the resolution to 100hz instead of 90hz then, using https://github.com/Nonary/documentation/wiki/DummyPlugs#4-rtss-limiter

**Comment by joemckie** [comment]
@Nonary I can see in this thread the refresh rate for the override is mentioned to be 90hz, but in the readme it says 60hz. Which one should it be? 😄

**Comment by DaLuke95** [comment]
I ended up switching from Sunshine to Apollo which includes a lot of scripts already - can only recommend!

**Comment by joemckie** [comment]
Ah nice - I’m mostly set up now with Sunshine. I was mostly just wondering if there was a reason for rendering at a lower frame rate than streaming at.

**Comment by Nonary** [comment]
> @Nonary I can see in this thread the refresh rate for the override is mentioned to be 90hz, but in the readme it says 60hz. Which one should it be? 😄

Honestly the refresh rate doesn't matter, in the readme I put it at 60hz because most displays will support that refresh rate without adding a custom resolution. As long as vsync is disabled and the game is running in borderless, the screen technically has an unlimited refresh rate at that point.

The caveat is that you need to frame cap the game, (using RTSS usually as I have a script for that) that way there isn't any frame pacing issues.

**Comment by Nonary** [comment]
This cannot be fixed from the script side, this is a windows specific issue whenever the resolution has not been added to the dropdown and the desktop and active signal modes must match exactly. Here is an example of a resolution added by NVIDIA that cannot be swapped to because it does not update the active signal mode. Although this might work if you have 24H2 installed

![Image](https://github.com/user-attachments/assets/d1e68604-e031-4d15-afaa-8767251b6b49)
